rfcs/0010-auto-execution-mode-selection.md

```diff
@@ -0,0 +1,94 @@
+---
+feature: Auto Execution Mode Selection
+authors:
+  - "DylanChen"
+start_date: "2022/11/03"
+---
+
+# Auto Execution Mode Selection
+
+## Summary
+
+This RFC introduces an auto execution mode selection feature for user's batch workloads so that OLTP queries can run in local mode and OLAP queries can run in distributed mode. As a result, users will experience a lower latency automatically.
+
+## Motivation
+
+Currently, our system exists 2 kind of execution mode: local and distributed. We want to run OLTP queries in local execution mode, while OLAP queries in distributed execution mode. However we leave the choice to users by `set query_mode = [local | distributed]` which in some way makes our distributed execution useless because most users don't know or understand what those options exactly mean. As a database, we have the ability and should make a good choice for users to reduce their tuning works. 
+
+## Design
+
+Since we don't have any statistics about table/mv, we are unable to calculate a meaningful cost for queries, so we focus on the oltp workload read pattern to decide what queries should run in local mode.
+
+### Query should run in local mode:
+
+- Single table primary key lookup. we can check our physical plan to ensure whether there is only one table scan in the plan tree and this table scan contains a point lookup scan range.
+- Single table two side bounded range scan. we can check our physical plan to ensure whether there is only one table scan in the plan tree and this table scan contains a two side bounded scan range.
+- Single table Index lookup. we can check our logical plan and physical plan to ensure there is only one table scan in the logical plan tree and there is a lookup join(non covering index) or there is a table scan in the physical plan tree(covering index).
```

Comment:
Another case is expression only sql, for example `select 1`

Reply:
Actually, `select` any expression without tables?